### PR TITLE
Koa 4975 swiftui component icon

### PR DESCRIPTION
**Pull request by frugoman**
Icon Component for SwiftUI and Also removing the usage of FONTS for the icons

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by frugoman** [comment]
To be rebased after the svgs PR is merged

File: Backpack-SwiftUI/Tests/Icons/BPKIconViewTests.swift [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@testable import Backpack_SwiftUI
+
+class BPKIconViewTests: XCTestCase {
+    func testLargeIcon() {
+        assertSnapshot(BPKIconView(.account, size: .large))
+    }
+    
+    func testSmallIcon() {
+        assertSnapshot(BPKIconView(.account, size: .small))
+    }
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Tests/Icons/BPKIconViewTests.swift:30]: the single icon tests seems too large? x2 size of what we would expect.

File: Backpack-SwiftUI/Tests/SnapshotHelpers/SnapshotHelpers.swift [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    testName: String = #function,
+    line: UInt = #line
+) {
+    let identifier: (UIUserInterfaceStyle) -> String = { mode in mode == .dark ? "dark-mode" : "light-mode" }
+    modes.forEach { mode in
+        let vc = UIHostingController(rootView: view)
+        
+        vc.overrideUserInterfaceStyle = mode
+        isRecording = false
+        assertSnapshot(matching: vc, as: .image, named:identifier(mode), file: file, testName: testName, line:line)
```
Comment (gert-janvercauteren) [inline Backpack-SwiftUI/Tests/SnapshotHelpers/SnapshotHelpers.swift:37]: you can pass in a trait in the as: .image(traits: <darkmodetrait here>)
I believe that is more stable than overriding the interfacestyle

File: Backpack/StarRating/Classes/BPKStar.m [hunk 3]
```diff
@@ -144,7 +144,7 @@ - (UIColor *)currentStarColor {
 - (BPKIconSize)iconSizeForStarSize:(BPKStarSize)size {
     switch (size) {
     case BPKStarSizeXLarge:
-        return BPKIconSizeXLarge;
+        return BPKIconSizeLarge;
```
Comment (gert-janvercauteren) [inline Backpack/StarRating/Classes/BPKStar.m:147]: Should we support the XL use-case? 
Right now setting XL to Large would not produce the wanted outcomes?

**Review by gert-janvercauteren — COMMENTED** [review]
We probably want to make sure it's ok to remove the XL icons for the star rating

**Review by gert-janvercauteren — APPROVED** [review]

**Pull request merged**